### PR TITLE
Add HighFive to externals

### LINKED ISSUE #5987: HighFive library for handling HDF5 Files
I am requesting a new external for handling HDF5 for generator level calculations.
This is a header-only library available at github:

https://github.com/BlueBrain/HighFive.git

Please install tag v2.0.

I have not tested with other versions.

### PR DESCRIPTION
**Pull request by mrodozov**
please test
Resolves: https://github.com/cms-sw/cmsdist/issues/5987

### CODE REVIEW

File: highfive-toolfile.spec [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+mkdir -p %{i}/etc/scram.d
+cat << \EOF_TOOLFILE >%{i}/etc/scram.d/highfive.xml
+<tool name="highfive" version="@TOOL_VERSION@">
+  <info url="https://github.com/BlueBrain/HighFive"/>
+  <client>
+    <environment name="HIGHFIVE_BASE" default="@TOOL_ROOT@"/>
+    <environment name="INCLUDE" default="$HIGHFIVE_BASE/include"/>
+  </client>
+  <runtime name="ROOT_INCLUDE_PATH" value="$INCLUDE" type="path"/>
+</tool>
```
Comment (smuzaffar) [inline highfive-toolfile.spec:20]: @mrodozov , please build locally and checkif highfix headers use `boost, hdf5 eigen or xtensor` then please make sure to add their dependency here too e..g 
```
<use name="boost"/> 
<use name="eigen"/>
```

Comment (mrodozov) [reply highfive-toolfile.spec:20]: It only had hdf5 and boost headers. I wonder why xtensor and eigen are in requirements then

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
@mrodozov , looks like `eigen and xtensor` are only used in tests ( which we have disabled here)
https://github.com/BlueBrain/HighFive/blob/master/CMakeLists.txt#L31-L32 . cmake also confirms this [a]. I would suggest to remove `eigen and xtensor` dependency. 

[a]
```
CMake Warning:
  Manually-specified variables were not used by the project:

    HIGHFIVE_USE_EIGEN
    HIGHFIVE_USE_XTENSOR
```

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
@mrodozov , do not forget to add the doc.txt file

**Comment by mrodozov** [comment]
It's there, but github doesn't show it, not sure why
https://github.com/cms-sw/cmsdist/tree/add-highfive - shows 4 commits, the PR shows 3 :/ 
I started it to see if the bot will get  it regardless, if not i'll squash it all in 1 commit and reopen the PR

**Comment by smuzaffar** [comment]
+externals

**Comment by smrenna** [comment]
Thank you for all of the hard work you do.    Clearly, this was harder than I thought it would be (because of the tests directory, which we don't really care about).

**Comment by smuzaffar** [comment]
@smrenna , yes integrating new externals takes a bit longer as we wantto make sure that we are correctly building it and toolfile definition for scram are correct :-)

This should be available in today's 23h IBs.

**Pull request merged**